backend/src/migrations/20240216181434_add_columns_to_user_details_table.ts

```diff
@@ -0,0 +1,53 @@
+import { type Knex } from 'knex';
+
+const TABLE_NAME = 'user_details';
+
+const ColumnName = {
+    AVATAR_URL: 'avatar_url',
+    USERNAME: 'username',
+    DATE_OF_BIRTH: 'date_of_birth',
+    WEIGHT: 'weight',
+    HEIGHT: 'height',
+    GENDER: 'gender',
+};
+
+const GENDER_ENUM = `${ColumnName.GENDER}_enum`;
+
+const Gender = {
+    MALE: 'male',
+    FEMALE: 'female',
+    OTHER: 'prefer not to say',
+};
+
+async function up(knex: Knex): Promise<void> {
+    await knex.schema.raw(
+        `CREATE TYPE ${GENDER_ENUM} AS ENUM ('${Gender.FEMALE}', '${Gender.MALE}', '${Gender.OTHER}');`,
+    );
```

Comment:
You can also create an enum like this:
`table.enum(ColumnName.GENDER, Object.values(Gender), {useNative: true, enumName: GENDER_ENUM_NAME});`
Instead of Gender object you could create just array of values and replace Object.values(Gender) with this array.
However, the choice is up to you, as both approaches will achieve the same result.

Reply:
ohh, thank you! i will keep this approach in the mind for the future